MazerPlatformer/GameObject.cs

```diff
@@ -162,57 +167,71 @@ public Component AddComponent(Component.ComponentType type, object value, string
             return component;
         }
 
-        private bool UpdateComponent(object newValue, Component found)
-        {
-            if (found == null) return false;
-            var oldValue = found.Value;
-            found.Value = newValue;
+        private Either<IFailure, object> UpdateComponent(object newValue, Component found)
+            => found == null
+                ? new NotFound($"Component not found to set value to {newValue}").ToFailure<Unit>()
+                : EnsureWithReturn(() =>
+                {
+                    var oldValue = found.Value;
+                    found.Value = newValue;
 
-            OnGameObjectComponentChanged?.Invoke(this, found.Id, found.Type, oldValue, newValue);
-            return true;
-        }
+                    OnGameObjectComponentChanged?.Invoke(this, found.Id, found.Type, oldValue, newValue);
+                    return newValue;
+                });
 
         public void AddState(State state) => States.Add(state);
 
         #region Diganostics
 
         // Draw the centre point of the object
-        protected void DrawCentrePoint(SpriteBatch spriteBatch) => DoIf(Diganostics.DrawCentrePoint, ()=> spriteBatch.DrawCircle(Centre, 2, 16, Color.Red, 3f));
-        
+        protected Either<IFailure, Unit> DrawCentrePoint(SpriteBatch spriteBatch)
+            => DoIff(Diganostics.DrawCentrePoint,
+                () => Ensure(() => spriteBatch.DrawCircle(Centre, 2, 16, Color.Red, 3f)))
+                .IgnoreFailure(Nothing);
```

Comment:
This is useful way of ignoring Failures so that they don't prevent the rest of the pipeline from working.